lib/powerpack/hash/map_keys.rb

```diff
@@ -0,0 +1,22 @@
+unless Hash.method_defined? :map_keys
+  class Hash
+    # Maps over the hash yielding the key-value pairs. The result of yielding
+    # is used as the value for the giving key in the returned hash.
+    #
+    # @return [Hash]
+    #
+    # @example
+    #   {}.map_keys # => {}
+    #
+    #   { first_name: 'john',  last_name: 'doe' }.map_keys { |_key, value|
+    #     value.capitalize
+    #   }
+    #
+    #   # => { first_name: 'John', last_name: 'Doe' }
+    #
+    #
+    def map_keys(&block)
+      Hash[map { |e| [e.first, yield(*e)] }]
```

Comment:
I'm struggling with the naming. This method, ideally, shouldn't be called anything other than `map`. `map_keys` may indeed be misleading, but `some_hash.map_values` may seem equivalent to `some_hash.values.map`, which is no the case.